lib/ui/addFeeder/AddFeederScreen.dart

```diff
@@ -0,0 +1,176 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:myportion_app/services/FirestoreUtils.dart';
+import 'package:myportion_app/services/helper.dart';
+import 'package:myportion_app/constants.dart';
+import 'package:myportion_app/model/Feeder.dart';
+import 'package:myportion_app/ui/FeederList/FeederListScreen.dart';
+
+class AddFeederScreen extends StatefulWidget {
+  final Feeder feeder;
+
+  AddFeederScreen({@required this.feeder});
+  @override
+  State createState() =>
+      _AddFeederScreenState(feeder, feeder.name, feeder.serialNum);
+}
+
+class _AddFeederScreenState extends State<AddFeederScreen> {
+  GlobalKey<FormState> _key = new GlobalKey();
+  AutovalidateMode _validate = AutovalidateMode.disabled;
+  Feeder feeder;
+  String feederName;
+  String serialNumber;
+
+  _AddFeederScreenState(this.feeder, this.feederName, this.serialNumber);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        elevation: 0.0,
+        backgroundColor: Color(COLOR_PRIMARY),
+        title: Text('Update Feeder'),
+        leading: IconButton(
+          icon: new Icon(Icons.arrow_back_ios, color: Colors.white),
+          onPressed: () async {
+            Feeder temp = await FireStoreUtils().getFeeder(feeder.id);
```

Comment:
After creating a feeder and entering a serial number, I received a null exception on this line. Not sure what exactly happened. Maybe we need to verify non-null before referencing `temp`?